ipaserver/install/server/install.py

```diff
@@ -761,12 +761,10 @@ def install(installer):
 
     # Create a directory server instance
     if not options.external_cert_files:
-        # Configure ntpd
+        # We have to sync time before generation of certificate and
+        # later ipa-server-uninstall will handle uninstallation
         if not options.no_ntp:
```

Comment:
I'm not sure where uninstall factors in here.

Reply:
I meant un-configuration (`restore_time_sync`) during ipa-server-install --uninstall:
```diff
diff --git a/ipaserver/install/server/install.py b/ipaserver/install/server/install.py
index 3a3b68792..564413a97 100644
--- a/ipaserver/install/server/install.py
+++ b/ipaserver/install/server/install.py
@@ -761,6 +761,13 @@ def install(installer):
 
     # Create a directory server instance
     if not options.external_cert_files:
+        # We have to sync time before certificate handling on master.
+        # As chrony configuration is moved from client here, unconfiguration of
+        # chrony will be handled here in uninstall() method as well by invoking
+        # the ipa-server-install --uninstall
+        if not options.no_ntp:
+            ipaclient.install.client.sync_time(options, fstore, sstore)
+
         if options.dirsrv_cert_files:
             ds = dsinstance.DsInstance(fstore=fstore,
                                        domainlevel=options.domainlevel,
@@ -906,7 +913,7 @@ def install(installer):
     try:
         args = [paths.IPA_CLIENT_INSTALL, "--on-master", "--unattended",
                 "--domain", domain_name, "--server", host_name,
-                "--realm", realm_name, "--hostname", host_name]
+                "--realm", realm_name, "--hostname", host_name, "--no-ntp"]
         if options.no_dns_sshfp:
             args.append("--no-dns-sshfp")
         if options.ssh_trust_dns:
@@ -1082,6 +1089,8 @@ def uninstall(installer):
         except Exception:
             pass
 
+    ipaclient.install.client.restore_time_sync(sstore, fstore)
+
     kra.uninstall()
 
     ca.uninstall()
```